src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/NotYetReady/If.txt

```diff
@@ -0,0 +1,56 @@
+// We don't evaluate the branch not taken. 
+>> If(5>10,1/0,99)
+99
+
+// We don't evaluate else if conditions until needed
+>> If(10>5,99,1/0=1,98)
+99
+
+>> If(true, If(false,1,2), 3)
+2
+
+>> If(true, 1)
+1
+
+>> If(false, 1)
+Blank()
+
+>> If(false, true)
+Blank()
+
+>> If(false, 1, 2)
+2
+
+>> If(true, 1, 2)
+1
+
+>> If(false, 1, true, 2)
+2
+
+>> If(false, 1, false, 2)
+Blank()
+
+>> If(false, 1, false, 2, 3)
+3
+
+>> If(false && true, false && true, false && true, false && true, false && true)
+false
+
+>> If(1 > 2 || 1 = 0 || 1 <> 1 || 2 < 1 || 3 <= 2 || 2 >= 3 || !true || (!false && false), "foo")
+Blank()
+
+// 'If' uses 1st arg as coercion target
+>> If(false, "3", 2)

```

Comment:
Consider also adding cases where the coercion results in an error, like
If(false, 2, "abcd")

Reply:
Thanks for the review @lesaltzm.  I will add the suggested cases. Work on this file is in progress.